core/tee/tee_rpmb_fs.c

```diff
@@ -437,12 +437,14 @@ static void tee_rpmb_free(struct tee_rpmb_mem *mem)
 	if (!mem)
 		return;
 
-	thread_rpc_free_payload(mem->phreq_cookie);
-	thread_rpc_free_payload(mem->phresp_cookie);
-	mem->phreq = 0;
-	mem->phreq_cookie = 0;
-	mem->phresp = 0;
-	mem->phresp_cookie = 0;
+	if (mem->phreq) {
```

Comment:
For the time being just clear the _cookie too when freeing.
Long term we should do something like option two, I'd like to think it through properly first.